### PR TITLE
feat(forms): be able to fill fields from the outside

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**

It's impossible to modify form properties from the outside without be annoyed with validation errors and sequencing issues.

**What is the chosen solution to this problem?**

Be able to give function to trigger to modify properties and errors.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/forms/stories-core/json/fields/core-resource-picker.json [hunk 1]
```diff
@@ -4,31 +4,29 @@
...
-      "placeholder": "Select an existing resource",
-      "multi": true
+      "key": "name",
+      "title": "Title change on picker change"
     },
     {
-      "key": "singleResourcePicker",
-      "title": "ResourcePicker without multi selection",
+      "key": "datasetId",
+      "title": "ResourcePicker that change the input above",
```
Comment (frassinier) [inline packages/forms/stories-core/json/fields/core-resource-picker.json]:
```suggestion
      "title": "ResourcePicker that changes the input above",
```

File: packages/forms/stories-core/json/fields/core-resource-picker.json [hunk 2]
```diff
@@ -4,31 +4,29 @@
...
   },
   "uiSchema": [
     {
-      "key": "multiResourcePicker",
-      "title": "ResourcePicker with multi selection",
-      "widget": "resourcePicker",
-      "placeholder": "Select an existing resource",
-      "multi": true
+      "key": "name",
+      "title": "Title change on picker change"
```
Comment (frassinier) [inline packages/forms/stories-core/json/fields/core-resource-picker.json]:
```suggestion
      "title": "Title changes on picker change"
```

File: packages/forms/src/UIForm/UIForm.container.test.js [hunk 3]
```diff
@@ -207,5 +207,55 @@ describe('UIForm container', () => {
...
+
+			const triggerPromise = instance.onTrigger();
+
+			triggerPromise.then(() => {
+				expect(instance.state.liveState.properties).toBe(properties);
+				done();
+			});
+		});
+
+		it('should call propertiers updater if given', done => {
```
Comment (frassinier) [inline packages/forms/src/UIForm/UIForm.container.test.js]:
```suggestion
		it('should call properties updater if given', done => {
```

File: packages/forms/src/UIForm/UIForm.container.js [hunk 4]
```diff
@@ -144,8 +144,21 @@ export default class UIForm extends React.Component {
...
 			}
+
+			if (data.properties) {
+				if (typeof data.properties === 'function') {
+					this.setState(setLiveStateProperties(data.properties(this.state.liveState.properties)));
+				} else {
+					this.setState(setLiveStateProperties(data.properties));
+				}
+			}
+
```
Comment (frassinier) [inline packages/forms/src/UIForm/UIForm.container.js]: I would prefer to read this, but it's not a blocker at all! 😏 

```javascript
const liveState = this.state.liveState;
if (data.errors) {
	let errors = data.errors;
	if (typeof data.errors === 'function') {
		errors = data.errors(liveState.errors);
	}
	this.setErrors(event, errors);
}
if (data.properties) {
	let properties = data.properties;
	if (typeof data.errors === 'function') {
		properties = data.properties(liveState.properties);
	}
	this.setState(setLiveStateProperties(properties));
}
```

File: packages/forms/src/UIForm/fields/ResourcePicker/ResourcePicker.component.js [hunk 5]
```diff
@@ -40,18 +42,30 @@ class ResourcePicker extends Component {
...
+			this.onTrigger(event, FILTER, { filters: this.state.filters })
 				.then(data => this.setState(data))
 				.finally(() => this.setState({ isLoading: false }));
 		});
 	}
 
+	onTrigger(event, eventName, payload) {
+		const { schema, properties, errors } = this.props;
+		const trigger = schema.triggers && schema.triggers.find(trig => trig.onEvent === eventName);
+		const formData = mutateValue(properties, schema, payload.value);
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/fields/ResourcePicker/ResourcePicker.component.js]: You don't need that anymore :)
It was one of our tries, but in fact you switched to a properties updater, so no need to have the up to date properties.

File: packages/forms/src/UIForm/fields/ResourcePicker/ResourcePicker.component.js [hunk 6]
```diff
@@ -67,8 +81,11 @@ class ResourcePicker extends Component {
 			selected.push(id);
 		}
 
+		const value = multi ? selected : selected[0];
 		this.setState({ filters: { ...this.state.filters, selected } });
-		this.onChange(event, multi ? selected : selected[0]);
+		this.onChange(event, value);
+
+		this.onTrigger(event, CHANGE, { value });
```
Comment (jsomsanith) [inline packages/forms/src/UIForm/fields/ResourcePicker/ResourcePicker.component.js]: Can you add a `triggers` part in the widget README please ?
- explain the 2 supported events
- a little example of how to use them (uischema, onTrigger callback)

Comment (jsomsanith) [reply packages/forms/src/UIForm/fields/ResourcePicker/ResourcePicker.component.js]: Please add a test for this.

File: packages/forms/src/UIForm/UIForm.container.js [hunk 7]
```diff
@@ -143,8 +143,20 @@ export default class UIForm extends React.Component {
...
+					errors = data.errors(liveState.errors);
+				}
+				this.setErrors(event, errors);
+			}
+			if (data.properties) {
+				let properties = data.properties;
+				if (typeof data.properties === 'function') {
+					properties = data.properties(liveState.properties);
+				}
+				this.setState(setLiveStateProperties(properties));
```
Comment (jmfrancois) [inline packages/forms/src/UIForm/UIForm.container.js]: I would have prefer a pacth system here.
The global properties is impossible to merge.
trigger may take time so the properties is clearly not in synch here

Comment (jsomsanith) [reply packages/forms/src/UIForm/UIForm.container.js]: The patch system is the if just above

**Pull request merged**